### PR TITLE
XIVY 12264 TLS.UpdateTo10.0

### PR DESCRIPTION
**Pull request by ivy-tgr**
- XIVY-12264 make tls-connection-tester compatible with version 10.0
- XIVY-12264 fix api changes & changed version in pom

### CODE REVIEW

File: tls-tester/pom.xml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <project xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 https://maven.apache.org/xsd/maven-4.0.0.xsd" xmlns="http://maven.apache.org/POM/4.0.0"
     xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
-  <modelVersion>4.0.0</modelVersion>
```
Comment (ivy-tgr) [inline tls-tester/pom.xml:4]: Is this the only version I need to raise ? @ivy-rew

Comment (ivy-rew) [reply tls-tester/pom.xml:4]: modelVersion must remain 4.0.0
but the version of the artifact must be raised from 8.0.0V > 10.0.0-SNAPSHOT

File: tls-tester/pom.xml [hunk 2]
```diff
@@ -1,7 +1,7 @@
 <?xml version="1.0" encoding="UTF-8"?>
-<project xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd" xmlns="http://maven.apache.org/POM/4.0.0"
+<project xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 https://maven.apache.org/xsd/maven-4.0.0.xsd" xmlns="http://maven.apache.org/POM/4.0.0"
     xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
-  <modelVersion>4.0.0</modelVersion>
+  <modelVersion>10.0.0</modelVersion>
   <groupId>com.axonivy.ivy.supplements</groupId>
   <artifactId>tls-connection-tester</artifactId>
   <version>8.0.0-SNAPSHOT</version>
```
Comment (ivy-rew) [inline tls-tester/pom.xml:7]: this version must be raised to 10.0.0-SNAPSHOT

File: tls-tester/pom.xml [hunk 3]
```diff
@@ -15,11 +15,10 @@
       <plugin>
         <groupId>com.axonivy.ivy.ci</groupId>
         <artifactId>project-build-plugin</artifactId>
-        <version>8.0.6</version>
+        <version>10.0.14</version>
```
Comment (ivy-rew) [inline tls-tester/pom.xml:18]: the CI build is not running. so you should first address that problem.
I think all you need is to change the java version to 17 .. maybe also the provider.
https://github.com/axonivy/tls-connection-tester/blob/master/.github/workflows/maven.yml#L25

File: tls-tester/src/com/axonivy/test/tls/TLSTest.java [hunk 4]
```diff
@@ -172,16 +171,10 @@ private void loadCustomClientKeystore()
...
-    if (!sslClientSettings.useCustomTrustStore())
-    {
-      data.addEntry("Custom client TrustStore set to NOT be used - skipping");
-      data.setResult(2);
-    } else
-    {
-      customTrustStore = loadKeyStore(sslClientSettings.getTrustStoreFile(), sslClientSettings.getTrustStorePassword()
-          .toCharArray(), sslClientSettings.getTrustStoreType(), sslClientSettings.getTrustStoreProvider(),
-          "CustomTrustStore", data);
-    }
```
Comment (ivy-rew) [inline tls-tester/src/com/axonivy/test/tls/TLSTest.java:184]: don't remove this feature right now; it's still valid if you upload your own keystore on the tls-tester dialog

**Review by ivy-rew — CHANGES_REQUESTED** [review]
fix the build ... delay feature removal

File: tls-tester/src/com/axonivy/test/tls/TLSTest.java (hunk 4 above)
Comment (ivy-tgr) [reply tls-tester/src/com/axonivy/test/tls/TLSTest.java:184]: The problem is that we no longer have the useCustomTrustStore method.

Comment (ivy-rew) [reply tls-tester/src/com/axonivy/test/tls/TLSTest.java:184]: ahh; you are right ... good then.

**Review by ivy-rew — APPROVED** [review]
Thanks; good work :+1:

**Pull request merged**